### PR TITLE
Fix Vercel build & deployment issue

### PR DESCRIPTION
**Pull request by iossocket**
# Fix Vercel build & deployment issue

## Types of change

- [ ] Feature
- [x] Bug
- [ ] Enhancement

### CODE REVIEW

**Review by Nadai2010 — CHANGES_REQUESTED** [review]
Hello @iossocket, `yarn vercel` and `yarn vercel:yolo` both commands are failing me.

imagen

**Comment by iossocket** [comment]
@Nadai2010  hmm... I tested on my side, and deployed on vercel. May I know your node version? thanks

**Comment by Nadai2010** [comment]
@iossocket mmm... I just tested with version 18.17.1 and 20.11.1. 

What could i be doing wrong? I also tried both times deleting the yarn.lock and running `yarn install`.

imagen

imagen

**Comment by metalboyrick** [comment]
hmmm no build issues on vercel? @Nadai2010 you can refer to the logs to see what errored out, perhaps it will help with trouble shooting.

**Comment by metalboyrick** [comment]
@Nadai2010 couldn't reproduce on my end works completely fine

image

here are the dependency versions that was in my system:
- Node.js - v20.10.0
- yarn - 3.2.3
- Vercel CLI - 33.7.1

File: packages/nextjs/next.config.mjs [hunk 1]
```diff
@@ -27,6 +29,11 @@ const nextConfig = {
   webpack: (config) => {
     config.resolve.fallback = { fs: false, net: false, tls: false };
     config.externals.push("pino-pretty", "lokijs", "encoding");
+    config.plugins.push(
+      new webpack.NormalModuleReplacementPlugin(/^node:(.*)$/, (resource) => {
+        resource.request = resource.request.replace(/^node:/, "");
+      }),
+    );
```
Comment (metalboyrick) [inline packages/nextjs/next.config.mjs:36]: just curious why we need this webpack extension ser?

**Review by metalboyrick — APPROVED** [review]
looks good to me and working fine

**Review by Nadai2010 — APPROVED** [review]
Everything is working, it must have been a minor mistake on my part. Thanks.

imagen

File: packages/nextjs/next.config.mjs (hunk 1 above)
Comment (iossocket) [reply packages/nextjs/next.config.mjs:36]: I found there is a module `node.url` can not be parsed correctly, so add this plugin to handle this case

**Pull request merged**